extensions/native/circuit/src/fri/mod.rs

```diff
@@ -483,35 +520,56 @@ impl<F: PrimeField32> InstructionExecutor<F> for FriReducedOpeningChip<F> {
             a: a_ptr_ptr,
             b: b_ptr_ptr,
             c: result_ptr,
-            d: addr_space,
-            e: length_ptr,
-            f: alpha_ptr,
+            d: alpha_ptr,
+            e: hint_id_ptr,
+            f: hint_offset_ptr,
+            g: ood_point_idx_ptr,
             ..
         } = instruction;
 
+        let addr_space = F::from_canonical_u32(AS::Native as u32);
         let alpha_read = memory.read(addr_space, alpha_ptr);
-        let length_read = memory.read_cell(addr_space, length_ptr);
+        // Assumption: b_ptr_ptr + 1 is the length of b.
```

Comment:
I don't think this is valid. An array in eDSL is just a pair (Ptr, Usize); they don't need to be adjacent in memory (and they aren't).

Reply:
Actually if you use the standard constructor, then ptr and len should be adjacent on the stack, but `len_ptr = ptr_ptr - 1` bc the stack grows downwards..